cypress/integration/conditions/date_conditions.cy.js

```diff
@@ -0,0 +1,305 @@
+            `${previousYear}-11-01`,
+            `${currentYear}-02-01`,
+            `${currentYear}-04-19`,
+        ])
+
+        assertChipContainsText(`${dimensionName}: 1 condition`)
+
+        assertTooltipContainsEntries([
+            stageName,
+            `Before or including: ${TEST_DATE}`,
+        ])
+    })
+
+    it('is empty / null', () => {
+        addConditions([
+            {
+                conditionName: 'is empty / null',
+            },
+        ])
+
+        expectTableToMatchRows([`${currentYear}-01-01`, `${currentYear}-03-01`])
+
+        assertChipContainsText(`${dimensionName}: 1 condition`)
+
+        assertTooltipContainsEntries([stageName, `Is empty / null`])
+    })
+
+    it('is not empty / not null', () => {
+        addConditions([
+            {
+                conditionName: 'is not empty / not null',
+            },
+        ])
+
+        expectTableToMatchRows([
+            `${previousYear}-12-10`,
+            `${previousYear}-12-11`,
+            `${previousYear}-11-15`,
+            `${previousYear}-11-01`,
+            `${currentYear}-01-01`,
+            `${currentYear}-01-03`,
+            `${currentYear}-02-01`,
+            `${currentYear}-04-19`,
+        ])
+
+        assertChipContainsText(`${dimensionName}: 1 condition`)
+
+        assertTooltipContainsEntries([stageName, `Is not empty / not null`])
+    })
+
+    it('2 conditions: after + before or including', () => {
+        const TEST_DATE_BFI = `${previousYear}-12-02`
+        const TEST_DATE_AFT = `${previousYear}-12-01`
+
+        addConditions([
+            {
+                conditionName: 'after',
+                value: TEST_DATE_AFT,
+            },
+            {
+                conditionName: 'before or including',
+                value: TEST_DATE_BFI,
+            },
+        ])
+
+        expectTableToMatchRows([`${previousYear}-12-11`])
+
+        assertChipContainsText(`${dimensionName}: 2 conditions`)
+
+        assertTooltipContainsEntries([
+            stageName,
+            `After: ${TEST_DATE_AFT}`,
+            `Before or including: ${TEST_DATE_BFI}`,
+        ])
+    })
+})
+
+describe('date types', () => {
+    const TEST_OPERATORS = [
+        'exactly',
+        'is not',
+        'after',
+        'after or including',
+        'before',
+        'before or including',
+        'is empty / null',
+        'is not empty / not null',
+    ]
+
+    const TEST_TYPES = [TEST_DIM_DATE, TEST_DIM_TIME, TEST_DIM_DATETIME]
+
+    TEST_TYPES.forEach((type) => {
+        it(`${type} has all operators`, () => {
+            cy.visit('/', EXTENDED_TIMEOUT)
+
+            selectEventProgram(ANALYTICS_PROGRAM)
+            openDimension(type)
+
+            cy.getWithDataTest('{button-add-condition}').click()
+            cy.contains('Choose a condition').click()
+
+            TEST_OPERATORS.forEach((operator) => {
+                cy.getBySel(
+                    'dhis2-uicore-select-menu-menuwrapper'
```

Comment:
not a big deal at all, but to increase test readability, would it be possible to set the dataTest value on this component? e.g. `conditions-menu`